riscv-sbi.adoc

```diff
@@ -1674,6 +1677,388 @@ The possible error codes returned in `sbiret.error` are shown in the
+                       │
+                       │
+                       │
+                       │
+                       │
+┌───────────┐          │
+│  Layer n  │          │
+├───────────┴────────────────────────────────┐
+│     Extend Measurements for layer n+1      │
+│ Generate evidence cert (CDI) for layer n+1 │
+│     Access-control current layer keys      │
+└────────────────────────────────────────────┘
+....
+
+The SBI implementation composes those measurements into an attestation evidence
+that can be used by attestation verifiers, either locally or remotely. Depending
+on the evidence verification results, a relying party may decide that the
+measured layered TCB are trustworthy and e.g. release and inject secrets into
+the platform.
+
+[#figure_attestation_flow]
+.TCB Attestation Flow
+....
+
+                                      ┌──────────────────┐
+                                      │ Reference Value  │
+                                      │     Provider     │
+                                      └────────┬─────────┘
+                                               │
+                                           Reference
+┌───────────┐                               Values
+│ Platform  │                                  │
+├───────────┴──────────┐                       │
+│   ┌────────────┐     │                       ▽
+│   │  Attester  │     │ Layered        ┌────────────┐
+│   │ (e.g. TVM) ├─────┼─Evidence──────▷│  Verifier  │
+│   └────────────┘     │                └──────┬─────┘
+│          △           │                       │
+│          │           │                       │
+│     Evidence for     │                  Attestation
+│       Attester       │                    Results
+│          │           │                       │
+│    ┌─────┴────┐      │                       │
+│    │ Layer 1  │      │                       │
+│    └──────────┘      │                       ▽
+│          △           │             ┌──────────────────┐
+│          │           │             │  Relying Party   │
+│ Evidence for layer 1 │             └──────────────────┘
+│          │           │                       │
+│    ┌─────┴─────┐     │                       │
+│    │  Layer 0  │     │                       │
+│    └───────────┘     │                       │
+└──────────────────────┘                       │
+            △                                  │
+            │                                  │
+            └─────────Secret Injection─────────┘
+
+....
+
+These functions provide the capabilities and features required to implement
+attestation for usages such as confidential computing, measured boot and
+verified boot.
+
+[#get-capabilities-function,reftext="Get Capabilities"]
+=== Function: Get Attestation and Measurement Capabilities (FID #0)
+
+[source, C]
+----
+struct sbiret sbi_attestation_get_capabilities(unsigned long *caps_addr
+                                               unsigned long caps_size);
+----
```

Comment:
Is it changed already ? Not sure if I am looking at the old PR.
It would also be good to prefix all the argument that (e.g caps_addr) updated by the SBI implementation with a prefix or suffix (e.g out) to indicate that.

Reply:
> Is it changed already ? Not sure if I am looking at the old PR.

You were looking at the right PR, I forgot to fix that one. Sorry.

> It would also be good to prefix all the argument that (e.g caps_addr) updated by the SBI implementation with a prefix or suffix (e.g out) to indicate that.

Good suggestion. I added a `_out` suffix for all pointer parameters that the SBI implementation updates.

Reply:
This needs to be marked as resolved.